### PR TITLE
RelaxNG validation with XInclude / File association report DOCTYPE error

### LINKED ISSUE #1421: RelaxNG validation with XInclude / File association report DOCTYPE error
Given :

 * `foo.rng`:
 
```xml
<grammar xmlns="http://relaxng.org/ns/structure/1.0"
	datatypeLibrary="http://www.w3.org/2001/XMLSchema-datatypes"
	xmlns:xi="http://www.w3.org/2001/XInclude">
	<start>
		<ref name="foo" />
	</start>
	<define name="foo">
		<element name="foo">
			<element name="bar">
				<attribute name="xml:base" />
			</element>
		</element>
	</define>
</grammar>
```

* `bar.xml`:

`<bar></bar>`

 * `foo.xml`
 
```xml
<foo xmlns:xi="http://www.w3.org/2001/XInclude">
	<xi:include href="bar.xml" />
</foo>
```

with : 

 * `XInclude enabled`
 * `foo.xml` associate with `foo.rng` file association

The is an error with DOCTYPE:

image

Although it should have no error.

### PR DESCRIPTION
**Pull request by angelozerr**
RelaxNG validation with XInclude / File association report DOCTYPE error

Fixes #1421

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/relaxng/xml/diagnostics/XMLFileAssociationRNCDiagnosticsTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+				"  <card>\r\n" + //
+				"    <name>Fred Bloggs</name>\r\n" + //
+				"    <email>fb@example.net</email>\r\n" + //
+				"  </card>\r\n" + //
+				"</addressBook>";
+		testDiagnosticsFor(xml, //
+				d(2, 5, 12, RelaxNGErrorCode.unknown_element), //
+				d(3, 5, 10, RelaxNGErrorCode.unexpected_element_required_element_missing));
+	}
+
```
Comment (JessicaJHee) [inline org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/relaxng/xml/diagnostics/XMLFileAssociationRNCDiagnosticsTest.java:67]: Are you able to add the `xinclude()` test you have in `XMLFileAssociationRNGDiagnosticsTest.java` here as well?

**Review by JessicaJHee — COMMENTED** [review]
Tested and works as expected! Just had one small question.

File: org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/relaxng/xml/diagnostics/XMLFileAssociationRNCDiagnosticsTest.java (hunk 1 above)
Comment (angelozerr) [reply org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/relaxng/xml/diagnostics/XMLFileAssociationRNCDiagnosticsTest.java:70]: Good catch!

I add the test with rnc.

Comment (JessicaJHee) [reply org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/relaxng/xml/diagnostics/XMLFileAssociationRNCDiagnosticsTest.java:70]: Thank you!

**Pull request merged**